src/ObjectStore/v1/Models/MetadataTrait.php

```diff
@@ -13,9 +13,12 @@ public function parseMetadata(ResponseInterface $response): array
         $metadata = [];
 
         foreach ($response->getHeaders() as $header => $value) {
-            if (0 === strpos($header, static::METADATA_PREFIX)) {
+            if (0 === stripos($header, static::METADATA_PREFIX)) {
                 $name            = substr($header, strlen(static::METADATA_PREFIX));
                 $metadata[$name] = $response->getHeader($header)[0];
+                if ($name !== strtolower($name)) {
+                    $metadata[strtolower($name)] = $response->getHeader($header)[0];
```

Comment:
I'm also not sure we need this at all. It might break the backward compatibility as swift metadata would have double keys.

Reply:
It allows my `$container->getMetadata()["temp-url-key"]` code to work indistinctly on both OVH swift 2.33 (and its all-lowercase headers) but also on another provider's Swift 2.25 with mixed-case headers' names.